osrf/mbzirc
# Add global service for retrieving wavefield params

Signed-off-by: Ian Chen <ichen@osrfoundation.org>

Added a service in `GameLogicPlugin` that provides wave gain and wave period parameters to `SurfacePlugin` and `WaveVisualPlugin`. This is needed so that we don't have to modify the `usv` model and all other vessel params when creating new worlds. Instead we can just specify them in one place.

The service is only available if a `<wavefield>` section is specified inside the `GameLogicPlugin` SDF params. When the `SurfacePlugin` is loaded, it uses this service to get the global wave params which will override its own values if the service exists.

To test:

Launch the `coast_cloud.sdf` world:

```
ros2 launch mbzirc_ros competition_local.launch.py ign_args:="-v 4 -r coast_cloud.sdf"
```

scroll up the console logs to where `SurfacePlugin` or `WaveVisualPlugin` is loaded and you should see this output. ~The last few lines indicate that wave params have been updated:~. The period and gain parameters should now be updated to the values specified in the `GamelogicPlugin`.

```
[ign gazebo-1] [Msg] Using Pierson-Moskowitz spectrum sampling wavefield model 
[ign gazebo-1] [Msg] Using Pierson-Moskowitz spectrum sampling wavefield model 
[ign gazebo-1] [Msg] Input Parameters:
[ign gazebo-1] [Msg] model:     PMS
[ign gazebo-1] [Msg] number:     3
[ign gazebo-1] [Msg] scale:      1.5
[ign gazebo-1] [Msg] angle:      0.4
[ign gazebo-1] [Msg] steepness:  0
[ign gazebo-1] [Msg] amplitude:  0
[ign gazebo-1] [Msg] period:     4.5
[ign gazebo-1] [Msg] direction:  1 0
[ign gazebo-1] [Msg] tau:  2
[ign gazebo-1] [Msg] gain:  0.32
[ign gazebo-1] [Msg] Derived Parameters:
[ign gazebo-1] [Msg] amplitudes:  0.0159463, 0.0819574, 0.0537983, 
[ign gazebo-1] [Msg] wavenumbers: 0.0716162, 0.161136, 0.362557, 
[ign gazebo-1] [Msg] omegas:      0.837758, 1.25664, 1.88496, 
[ign gazebo-1] [Msg] periods:     [Msg] 7.5, [Msg] 5, [Msg] 3.33333, [Msg] 
[ign gazebo-1] [Msg] phases:      0, 0, 0, 
[ign gazebo-1] [Msg] steepnesses: 0, 0, 0, 
[ign gazebo-1] [Msg] directions:  [Msg] 0.921061 -0.389418; [Msg] 1 0; [Msg] 0.921061 0.389418; [Msg] 
[ign gazebo-1] [Msg] Using Pierson-Moskowitz spectrum sampling wavefield model 
```
